RELEASE-NOTES.md

```diff
@@ -1,5 +1,42 @@
 # Prefect Release Notes
 
+## Release 2.14.0
+
+### Introducing the `prefect-client`
+
+This release provides a new way of running flows using the `prefect-client` package. This slimmed down version of `prefect` has a small surface area of functionality and is intended for interacting with Prefect-server and Prefect-cloud **only**. You can install `prefect-client` by using `pip`:
+
```

Comment:
We should just say "the Prefect server" like in our docs. I'd defer to @discdiver on how we speak to that

Reply:
I don't know that we need mention Prefect Cloud here

Reply:
Is this because we don't want to make a distinction between the two and we generally default to saying "Prefect server"?

Reply:
it's because this is the open source release I think. But also yeah, "the Prefect server" is a catch-all